domain/model/state/controllerstate_test.go

```diff
@@ -1449,6 +1449,56 @@ UPDATE model SET life_id = 1 WHERE uuid = ?
 	})
 }
 
+// TestGetEmptyCredentialsModel verifies that we can retrieve the correct model
+// information even when the model credentials are empty, for both IAAS and Container model types.
```

Comment:
Thanks @CodingCookieRookie, can you expand on this comment a little more for me please?

Be explicit in what we are trying to test. That is we want to make sure the model view continues to work when a credential is nil and potential a join in the sql stops working.

It easy for us to remember why we did this today but in a few years it might not be so clear to the reader why this test exists.

Reply:
Sure thing thanks tom :)